torch/nn/modules/module.py

```diff
@@ -249,26 +249,41 @@ def forward(self, x):
     the change."""
 
     training: bool
+    _parameters: Dict[str, Optional[Parameter]]
+    _buffers: Dict[str, Optional[Tensor]]
+    _non_persistent_buffers_set: Set[str]
+    _backward_hooks: Dict[int, Callable]
     _is_full_backward_hook: Optional[bool]
+    _forward_hooks: Dict[int, Callable]
+    _forward_pre_hooks: Dict[int, Callable]
+    _state_dict_hooks: Dict[int, Callable]
+    _load_state_dict_pre_hooks: Dict[int, Callable]
+    _load_state_dict_post_hooks: Dict[int, Callable]
+    _modules: Dict[str, Optional['Module']]
 
     def __init__(self) -> None:
         """
         Initializes internal Module state, shared by both nn.Module and ScriptModule.
+
+        Calls object.__setattr__(self, 'a', a) instead of the typical self.a = a
+        to avoid Module.__setattr__ overhead. Module's __setattr__ has special
+        handling for parameters, submodules, and buffers but simply calls into
+        object.__setattr__ for all other attributes.
         """
         torch._C._log_api_usage_once("python.nn_module")
 
-        self.training = True
-        self._parameters: Dict[str, Optional[Parameter]] = OrderedDict()
-        self._buffers: Dict[str, Optional[Tensor]] = OrderedDict()
-        self._non_persistent_buffers_set: Set[str] = set()
-        self._backward_hooks: Dict[int, Callable] = OrderedDict()
-        self._is_full_backward_hook = None
-        self._forward_hooks: Dict[int, Callable] = OrderedDict()
-        self._forward_pre_hooks: Dict[int, Callable] = OrderedDict()
-        self._state_dict_hooks: Dict[int, Callable] = OrderedDict()
-        self._load_state_dict_pre_hooks: Dict[int, Callable] = OrderedDict()
-        self._load_state_dict_post_hooks: Dict[int, Callable] = OrderedDict()
-        self._modules: Dict[str, Optional['Module']] = OrderedDict()
+        object.__setattr__(self, 'training', True)
```

Comment:
nit: `super().__setattr__()` throughout

Reply:
Pre-existing behavior in module's `__setattr__` calls `object.__setattr__` directly, so I think we should either leave these as they are for consistency or change it there as well. What do you think?

https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L1256

Reply:
I'd vote to change to `super()` throughout as it's good practice. @albanD wdyt?

Reply:
Agreed that we should use `super()` unless we have a good reason not to do it.